zallet/src/components/sync.rs

```diff
@@ -389,7 +389,21 @@ async fn recover_history(
                     &from_state,
                     scan_range.len(),
                 )
-            })?;
+            });
+
+            // If a scan error is encountered, truncate to recover.
+            match scan_result {
+                Ok(_) => {}
+                Err(chain::error::Error::Scan(ScanError::PrevHashMismatch { at_height })) => {
+                    db_data
+                        .truncate_to_height(at_height - 10)
```

Comment:
This will break the internal state of the `steady_state` task. And given that it's supposed to be the `steady_state` task that handles reorgs, that suggests there is a different problem causing this bug. My current suspicion is unclean shutdown (I've seen this in a few different ways, such as #136).